File: NcmPlayer/Framework/Model/Album.cs
namespace NcmPlayer.Framework.Model
{
    public class Album
    {
        /// <summary>
        /// 专辑Id
        /// </summary>
        public int Id
        {
            get; set;
        }

        /// <summary>
        /// 专辑名称
        /// </summary>
        public string Name
        {
            get; set;
        }

        /// <summary>
        /// 专辑封面
        /// </summary>
        public string PicUrl
        {
            get; set;
        }

        /// <summary>
        /// 发行时间
        /// </summary>
        public DateTime CreateDate
        {
            get; set;
        }

        /// <summary>
        /// 描述
        /// </summary>
        public string Description
        {
            get; set;
        }

        /// <summary>
        /// 专辑歌曲
        /// </summary>
        public List<Music> Musics
        {
            get; set;
        }

        /// <summary>
        /// 专辑作者
        /// </summary>
        public List<Artist> Artists
        {
            get; set;
        }

        /// <summary>
        /// 专辑作者
        /// </summary>
        public string ArtistsName
        {
            get
            {
                if (Artists == null) return string.Empty;
                return string.Join("/", Artists.Select(x => x.Name));
            }
        }

        /// <summary>
        /// 收藏数目
        /// </summary>
        public int CollectionCount
        {
            set; get;
        }

        /// <summary>
        /// 歌曲数目
        /// </summary>
        public int TrackCount
        {
            get; set;
        }
    }
}

File: src-tauri/src/models/album.rs
use serde::{Deserialize, Serialize};

use super::{artist::Artist, song::Song};

#[derive(Serialize, Deserialize, Clone, Debug)]
#[serde(rename_all = "camelCase")]
pub struct Album {
    pub id: String,
    pub md5: String,
    pub name: String,
    pub share_url: String,
    pub avatar_url: String,
    pub small_avatar_url: String,
    pub middle_avatar_url: String,
    pub create_date: String,
    pub description: String,
    pub songs: Vec<Song>,
    pub artists: Vec<Artist>,
    pub artists_name: String,
    pub adapter_slug: String,
    pub collection_count: u32,
    pub track_count: u32,
}

impl Album {
    pub fn empty() -> Self {
        Self {
            id: String::new(),
            md5: String::new(),
            name: String::new(),
            share_url: String::new(),
            avatar_url: String::new(),
            small_avatar_url: String::new(),
            middle_avatar_url: String::new(),
            create_date: String::new(),
            description: String::new(),
            songs: vec![],
            artists: vec![],
            artists_name: String::new(),
            adapter_slug: String::new(),
            collection_count: 0,
            track_count: 0,
        }
    }
}
